### PR TITLE
Normalize external inline link whitespace in one pass

### PR DESCRIPTION
**Pull request by dahlia**
Fixes #28.

Why this is needed
------------------

Hongdown turns external inline links into reference links.  Before this change, the displayed text and the generated label followed different whitespace paths.  For `[a  b](https://example.com/)`, the first run retained both spaces in the visible text but wrote a definition using the normalized label.  Parsing that output collapsed the link text, so a second run changed it again.  A formatter should reach its stable form on the first run.

How the serializer keeps the right distinctions
-----------------------------------------------

The serializer now walks the inline nodes inside a link and normalizes only ordinary text.  Runs of CommonMark ASCII whitespace become one space.  Code spans, math, inline HTML, hard breaks, and non-ASCII whitespace retain their contents.  HTML element depth is tracked through emphasis and strong spans, so text inside inline HTML is not mistaken for ordinary Markdown text.

Images are handled as complete inline constructs during the same walk. Recursive image detection keeps image-bearing links inline even when the image is nested under emphasis or strong spans, or when the link appears in a heading.  This prevents image syntax from becoming an invalid reference label. In headings, soft breaks become spaces before inline link text is emitted, so the serializer's internal marker cannot reach the output.

Normalizing ordinary text before reference generation gives the displayed text and its label stable ASCII spacing on the first pass.  Separate label normalization can still prepare preserved inline syntax for CommonMark lookup without altering its display.  *STYLE.md* records this boundary.

How this was verified
---------------------

Each edge case was first captured by a failing regression test in *src/serializer/tests.rs*.  The tests cover doubled spaces and tabs, mixed ordinary and whitespace-sensitive content, non-ASCII whitespace, nested images, image-bearing links in headings, and multiline heading links.  They also format the results again to check that the first output is already stable.

The full Rust, WASM, and VS Code test suites pass through `mise run test`. Formatting, Clippy, type checking, Markdown checking, the WASM build, and VS Code type checking pass through `mise run check`.

### CODE REVIEW

File: src/serializer/tests.rs [hunk 1]
```diff
@@ -7460,6 +7486,99 @@ fn test_generated_label_normalization_preserves_link_content() {
...
+            result.output, expected,
+            "failed for U+{:04X}",
+            whitespace as u32
+        );
+        assert!(result.warnings.is_empty());
+
+        let second = parse_and_serialize_with_warnings(&result.output);
+        assert_eq!(second.output, result.output);
+        assert!(second.warnings.is_empty());
+    }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply src/serializer/tests.rs:7546]: Declined: CommonMark defines NBSP and U+3000 as Unicode whitespace and collapses internal whitespace when normalizing reference labels, so the explicit `[a b]` label correctly matches the `[a b]` definition. The regression test reparses that output without an undefined-reference warning and confirms a stable second pass.

File: src/serializer/inline.rs [hunk 2]
```diff
@@ -24,6 +29,181 @@ impl<'a> Serializer<'a> {
         text
     }
 
+    pub(super) fn contains_image<'b>(node: &'b AstNode<'b>) -> bool {
+        node.children().any(|child| {
+            matches!(&child.data.borrow().value, NodeValue::Image(_)) || Self::contains_image(child)
+        })
```
[the remark this answers is not in the document]
Comment (dahlia) [reply src/serializer/inline.rs:35]: Addressed in 634d3f7: *STYLE.md* now documents the inline exception for image-bearing external links, including nested images, and explains why their bracketed syntax cannot form a valid reference label.

File: src/serializer/inline.rs [hunk 3]
```diff
@@ -24,6 +29,181 @@ impl<'a> Serializer<'a> {
...
+                            has_content,
+                        );
+                        output.push_str(&text);
+                        has_content = true;
+                    }
+                }
+            }
+        }
+
+        output.push_str(&pending_whitespace);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply src/serializer/inline.rs]: Addressed in 080386d: pending CommonMark ASCII whitespace is now flushed through the same single-space path at both link-text edges, with leading, trailing, and trailing-tab regression coverage.

File: CHANGES.md [hunk 4]
```diff
@@ -6,6 +6,12 @@ Version 0.5.2
 
 To be released.
 
+ -  Fixed a bug where an external inline link whose text contained consecutive
```
[the remark this answers is not in the document]
Comment (dahlia) [reply CHANGES.md:9]: Declined: every commit in this PR already includes an `Assisted-by: Codex:gpt-5.6-sol` trailer. GitHub shows the trailer on a089979, 634d3f7, and 080386d, with the initial commit also crediting Claude Code.
